test/NetTopologySuite.Tests.NUnit/Geometries/GeometryOverlayTest.cs

```diff
@@ -0,0 +1,115 @@
+        }
+
+        [Test]
+        public void TestOverlayNGFixed()
+        {
+            //GeometryOverlay.setOverlayImpl(GeometryOverlay.OVERLAY_PROPERTY_VALUE_NG);
+            NtsGeometryServices.Instance = new NtsGeometryServices {GeometryOverlay = GeometryOverlay.NG};
+            var pmFixed = new PrecisionModel(1);
+            var expected = Read("POLYGON ((1 2, 4 1, 1 1, 1 2))");
+
+            CheckIntersectionPM(pmFixed, expected);
+        }
+
+        [Test]
+        public void TestOverlayNGFloat()
+        {
+            //GeometryOverlay.setOverlayImpl(GeometryOverlay.OVERLAY_PROPERTY_VALUE_NG);
+            NtsGeometryServices.Instance = new NtsGeometryServices {GeometryOverlay = GeometryOverlay.NG};
+            var pmFloat = new PrecisionModel();
+            var expected = Read("POLYGON ((1 1, 1 2, 4 1.25, 4 1, 1 1))");
+
+            CheckIntersectionPM(pmFloat, expected);
+        }
+
+        private void CheckIntersectionPM(PrecisionModel pmFixed, Geometry expected)
+        {
+            var geomFactFixed = new GeometryFactory(pmFixed);
+            Assert.That(geomFactFixed.GeometryOverlay, Is.EqualTo(NtsGeometryServices.Instance.GeometryOverlay));
+
+            var a = Read(geomFactFixed, "POLYGON ((1 1, 1 2, 5 1, 1 1))");
+            var b = Read(geomFactFixed, "POLYGON ((0 3, 4 3, 4 0, 0 0, 0 3))");
+            var actual = a.Intersection(b);
+            CheckEqual(expected, actual);
+        }
+
+        [Test]
+        public void TestOverlayOld()
+        {
+            // must set overlay method explicitly since order of tests is not deterministic
+            //GeometryOverlay.setOverlayImpl(GeometryOverlay.OVERLAY_PROPERTY_VALUE_OLD);
+            NtsGeometryServices.Instance = new NtsGeometryServices {GeometryOverlay = GeometryOverlay.Old};
+            CheckIntersectionFails();
+        }
+
+        [Test]
+        public void TestOverlayNG()
+        {
+            //GeometryOverlay.setOverlayImpl(GeometryOverlay.OVERLAY_PROPERTY_VALUE_NG);
+            NtsGeometryServices.Instance = new NtsGeometryServices {GeometryOverlay = GeometryOverlay.NG};
+            CheckIntersectionSucceeds();
+        }
+
+        private void CheckIntersectionFails()
+        {
+            try
+            {
+                TryIntersection();
+                Assert.Fail("Intersection operation should have failed but did not");
+            }
+            catch (TopologyException ex)
+            {
+                // ignore - expected result
+            }
+        }
+
+        private void CheckIntersectionSucceeds()
+        {
+            try
+            {
+                TryIntersection();
+            }
+            catch (TopologyException ex)
+            {
+                Assert.Fail("Intersection operation failed.");
+            }
+        }
+
+        private void TryIntersection()
+        {
+            var a = Read(NtsGeometryServices.Instance.CreateGeometryFactory(),
```

Comment:
suggestion: add a comment here to find a different failure case if something changes that breaks our "this should fail" test.